assets/js/components/user-input/UserInputPreview.js

```diff
@@ -115,33 +105,6 @@ export default function UserInputPreview( props ) {
 		return <ProgressBar />;
 	}
 
-	const updateErrorMessages = () => {
-		const newErrorMessages = USER_INPUT_QUESTIONS_LIST.reduce(
-			( errors, slug ) => {
-				const errorMessage = getErrorMessageForAnswer(
-					settings?.[ slug ]?.values || [],
-					USER_INPUT_MAX_ANSWERS[ slug ]
-				);
-				errors[ slug ] = errorMessage;
-				return errors;
-			},
-			{}
-		);
-
-		setErrorMessages( newErrorMessages );
-
-		return newErrorMessages;
-	};
-
-	const onSaveClick = () => {
-		const newErrorMessages = updateErrorMessages();
-
-		const hasErrors = Object.values( newErrorMessages ).some( Boolean );
-		if ( ! hasErrors ) {
-			submitChanges();
-		}
-	};
```

Comment:
As per above, this dance is no longer needed, as the save button will be disabled if there is any error.